appformer-kogito-bridge/src/test/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourCustomSelectorPositionProviderTest.java

```diff
@@ -0,0 +1,101 @@
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.appformer.kogito.bridge.client.guided.tour;
+
+import java.util.Objects;
+
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import elemental2.dom.Console;
+import elemental2.dom.DomGlobal;
+import org.appformer.kogito.bridge.client.guided.tour.service.api.Rect;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class GuidedTourCustomSelectorPositionProviderTest {
+
+    @Mock
+    private Console console;
+
+    private GuidedTourCustomSelectorPositionProvider positionProvider;
+
+    @Before
+    public void setup() {
+        positionProvider = spy(GuidedTourCustomSelectorPositionProvider.getInstance());
+        DomGlobal.console = console;
+    }
+
+    @Test
+    public void testGetPosition() {
+        final Rect rect1 = makeRect(1);
+        final Rect rect2 = makeRect(2);
+
+        positionProvider.registerPositionProvider("TEST_PROVIDER_1", name -> Objects.equals(name, "OBJECT-1") ? rect1 : rect2);
+
+        assertEquals(rect1, positionProvider.getPosition("TEST_PROVIDER_1:::OBJECT-1"));
+        assertEquals(rect2, positionProvider.getPosition("TEST_PROVIDER_1:::OBJECT-2"));
+    }
+
+    @Test
+    public void testGetPositionWhenSelectorIsInvalid() {
+        final Rect rect1 = makeRect(1);
+        final Rect rect2 = makeRect(2);
+
+        positionProvider.registerPositionProvider("TEST_PROVIDER_2", name -> Objects.equals(name, "OBJECT-1") ? rect1 : rect2);
+
+        final Rect position = positionProvider.getPosition("TEST_PROVIDER_2___OBJECT-1");
+
+        verify(console).warn("[Guided Tour - Position Provider] Invalid custom query selector: TEST_PROVIDER_2___OBJECT-1");
+        assertNotNull(position);
+    }
+
+    @Test
+    public void testGetPositionWhenNoSelectorIsRegistered() {
+        final Rect position = positionProvider.getPosition("TEST_PROVIDER_3:::OBJECT-1");
+
+        verify(console).warn("[Guided Tour - Position Provider] The position provider could not be found: TEST_PROVIDER_3");
+        assertNotNull(position);
+    }
+
+    @Test
+    public void testGetPositionWhenSelectorIsNull() {
+        final Rect position = positionProvider.getPosition(null);
+
+        verify(console).warn("[Guided Tour - Position Provider] Invalid custom query selector: null");
+        assertNotNull(position);
```

Comment:
Shouldn't we check if `position` is `Rect.NONE`?

Reply:
I've improved tests on this class. Unfortunately, `Rect` is a native type so we have some assertion limitations, but I believe this test class is good enough now.